src/Legacy/Storage.php

```diff
@@ -401,10 +401,10 @@ public function saveContent(Content $content, $comment = null)
         }
 
         // Test to see if this is a new record, or an update
-        if (empty($fieldvalues['id'])) {
-            $create = true;
-        } else {
-            $create = false;
+        $create = empty($fieldvalues['id']);
+
+        if (!isset($fieldvalues['slug'])) {
+            $fieldvalues['slug'] = ''; // Prevent 'slug may not be NULL'
```

Comment:
ah @GawainLynch. Definitely no problem. I know you have a lot to do and it's ok for me to wait.
Thank you for the kind words.

I know the slug should somehow be present and I'm not into the code at all anymore (maybe because of 🍺 ), but as far as I understand the slug is now required to save the record. At least in my case.

This means that there are two possibilities right now (if the user has no slug the content type):
1. create a slug (silently) in the background
2. update legacy code to work without slug.

When I interpret your answer correctly, only 1 can prevent BC breaks because 2 is too fragile.

All in all it sounds like a too big topic for this PR, isn't it?

Please let me know what I can contribute or how I can help.
Because a generated slug in the content type is the short fix for us at the moment and we are fine with that.

Thank you for your fast replies. 🥇